pkg/check/precheck.go

```diff
@@ -0,0 +1,126 @@
+package check
+
+import (
+	"sync"
+
+	"github.com/juju/errors"
+	"github.com/ngaut/log"
+)
+
+// Checker is interface that defines checker.
```

Comment:
`// Checker is the interface that defines checker.`
and some comments below ditto